decidim-core/lib/tasks/upgrade/decidim_fix_nickname_uniqueness.rake

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+namespace :decidim do
+  namespace :upgrade do
+    desc "Modify nicknames with random numbers when exists similar ones case insensitively"
+    task fix_nickname_uniqueness: :environment do
+      logger = Logger.new($stdout)
+      logger.info("Updating users nickname ...")
+
+      # list of users already changed in the process
+      has_changed = []
+
+      Decidim::User.find_each do |user|
+        next if has_changed.include? user
```

Comment:
If we change the array to contain the IDs only, this also needs to be changed to:
```suggestion
        next if has_changed.include? user.id
```